spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -145,88 +145,134 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
   // CAST from ByteType
 
   test("cast ByteType to BooleanType") {
-    castTest(generateBytes(), DataTypes.BooleanType)
+    val hasIncompatibleType = CometSparkSessionExtensions.isComplexTypeReaderEnabled(
+      conf) && !CometConf.COMET_SCAN_ALLOW_INCOMPATIBLE.get()
```

Comment:
This code block is repeated many times. Maybe it could be moved out to a val in the class instead? Perhaps something like:

```scala
  lazy val usingDataFusionParquetReader: Boolean = {
    CometSparkSessionExtensions.isComplexTypeReaderEnabled(
      conf) && !CometConf.COMET_SCAN_ALLOW_INCOMPATIBLE.get()
  }

  test("cast ByteType to BooleanType") {
    castTest(generateBytes(), DataTypes.BooleanType, hasIncompatibleType = usingDataFusionParquetReader)
  }
```

Reply:
Done